.github/workflows/ci.yaml

```diff
@@ -97,13 +119,45 @@ jobs:
           path: |
             ~/.cargo/
             target/
+            .venv/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Python Bindings Tests
         run: |
-          just python build
           just python test
 
+  benchmarks:
+    name: Benchmarks (Scale Factor = ${{ matrix.scale_factor }})
+    runs-on: ubuntu-latest-8-cores
+    needs: ["build-python"]
```

Comment:
we could make sure that this doesn't run unless all the other tests pass? 

adding extra dependencies would make things slower but no one is really waiting on this, so maybe it'd be better to not have to sometimes throw out data later if failed?

these are legit questions, I think both answers have merit (and it might make sense to do development with fewer dependencies and then add  them right before we're ready to merge), but should definitely answer them.

Reply:
That makes sense if we're running benchmarks on each commit. These are only run if there's a merge to the main (which should only happen if the tests pass). That's a whole different conversation, though.

For now, this should work fine. We'll have to add a new job for running micro-benchmarks for each commit (when we do).